### ISSUE TITLE
Tracking Issue for multiple Undo stack changes

### ISSUE THREAD

**Issue by KeyWorksRW** [KeyWorksRW/wxUiEditor#1051]
# Tracking Issue

This is a tracking issue -- the list at the bottom refers to multiple issues that all 
need a resolution. Once all items are checked, this issue can be closed.

## Background

The tool for modifying custom ids with user-defined prefixes and suffixes (#1017) is going to run into a similar problem that that auto_update property of Images Lists has encounter. Changes are being made to multiple nodes throughout the project, and the `UndoAction` classes that handle this need to both Undo and Redo all the changes at once.

This issue is to track changes to our Undo functionality that should be completed before tackling the bug in auto_update undo, and the new functionality of the custom ID tool. The following changes need to be made:

## List of Changes

- [x] Track Undo stack memory use (#1030) -- this is for testing to ensure we're not running into memory issues with large UndoActions sitting on the Undo stack.
- [ ] Add an Undo max history -- there is no value to having hundres of UndoActions on the stack since no user is going to perform hundreds of Undos. A history limit should help with running into memory issues
- [ ] We need to look closer at the memory cost of using GroupUndoActions versus storing a parent Node. It _should_ be more efficient, but I would prefer to be sure.
- [x] SortProjectAction has the largest addition to the Undo stack by storing the entire project onto the stack. That will be handy while tracking Undo memory use, but this needs an overhaul to reduce the memory footprint, and maybe see if we can't come up with a better mechanism for handling large node changes.
- [ ] Figuring out how to fix the AutoImagesAction (#1043) Undo bug will be critical before adding the custom id editor which like AutoImages will affect a large number of nodes.

**Comment by Randalphwa** [comment]
The Sort Project command does indeed take up a _lot_ of memory on the Undo/Redo stack, but I can't think of a way to avoid that -- we have to make a copy of every Node in order to preserve the parent/child connection. That means a copy of the entire Project has to be saved.

**Comment by Randalphwa** [comment]
The Global Custom IDs Editor only puts property changes on the Undo stack, so it's quite small. We still need to deal with the broken `AutoImagesAction` however.

**Comment by KeyWorksRW** [comment]
auto_update has been postponed until 1.2, and we do _not_ have time to add an Undo max history before the upcoming release,
so postponing.

**Comment by KeyWorksRW** [comment]
Incoming PR will remove the AutoUpdate functionality, and instead uses a new auto_add property that doesn't have all the problems associated with the AutoUpdate.
